plugin/src/main/java/org/opensearch/ml/model/MLModelCacheHelper.java

```diff
@@ -92,6 +92,7 @@ public synchronized void initModelStateLocal(
         modelCache.setDeployToAllNodes(false);
         modelCache.setLastAccessTime(Instant.now());
         modelCaches.put(modelId, modelCache);
+        setIsAutoDeploying(modelId, true);
```

Comment:
Looks `setIsAutoDeploying` method can be removed to simplify. Just need to add `modelCache.setIsAutoDeploying(true)` after line 93 ?